src/modules/fundingManager/depositVault/FM_DepositVault_v1.sol

```diff
@@ -0,0 +1,115 @@
+pragma solidity 0.8.23;
+
+// Internal Interfaces
+import {IOrchestrator_v1} from
+    "src/orchestrator/interfaces/IOrchestrator_v1.sol";
+import {IFundingManager_v1} from "@fm/IFundingManager_v1.sol";
+import {IFM_DepositVault_v1} from
+    "@fm/depositVault/interfaces/IFM_DepositVault_v1.sol";
+
+// Internal Dependencies
+import {Module_v1} from "src/modules/base/Module_v1.sol";
+
+import {Initializable} from "@oz-up/proxy/utils/Initializable.sol";
+
+// External Interfaces
+import {IERC20} from "@oz/token/ERC20/extensions/IERC20Metadata.sol";
+
+// External Dependencies
+import {ERC165Upgradeable} from
+    "@oz-up/utils/introspection/ERC165Upgradeable.sol";
+
+// External Libraries
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title   Deposit Vault Funding Manager
+ *
+ * @notice  This contract allows users to deposit tokens to fund the workflow.
+ *
+ * @dev     Implements {IFundingManager_v1} interface.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to our Security Policy
+ *                          at security.inverter.network or email us directly!
+ *
+ * @author  Inverter Network
+ */
+contract FM_DepositVault_v1 is
+    IFundingManager_v1,
+    IFM_DepositVault_v1,
+    Module_v1
+{
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(Module_v1)
+        returns (bool)
+    {
+        return interfaceId == type(IFundingManager_v1).interfaceId
+            || interfaceId == type(IFM_DepositVault_v1).interfaceId
+            || super.supportsInterface(interfaceId);
+    }
+
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // Storage
+
+    /// @dev The token that is deposited.
+    IERC20 private _token;
+
+    /// @dev Storage gap for future upgrades
+    uint[50] private __gap;
+
+    //--------------------------------------------------------------------------
+    // Init Function
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata,
+        bytes memory configData
+    ) external override initializer {
+        __Module_init(orchestrator_, metadata);
+
+        address orchestratorTokenAddress = abi.decode(configData, (address));
+        _token = IERC20(orchestratorTokenAddress);
+    }
+
+    //--------------------------------------------------------------------------
+    // Public View Functions
+
+    /// @inheritdoc IFundingManager_v1
+    function token() public view returns (IERC20) {
+        return _token;
+    }
+
+    //--------------------------------------------------------------------------
+    // Public Mutating Functions
+
+    /// @inheritdoc IFM_DepositVault_v1
+    function deposit(uint amount) external {
```

Comment:
btw, just noticing this function in the FM_Rebasing_v1
https://github.com/InverterNetwork/contracts/blob/dev/src/modules/fundingManager/rebasing/FM_Rebasing_v1.sol#L150

Reply:
Do we really need it in this case? The FundingManager does not track balances, so you cannot deposit "for" somebody. In the other cases the depositFor/buyFor takes funds from the sender and increases the balance of somebody else, but here the implementation of a buyFor function would look exactly the same...

Reply:
> The FundingManager does not track balances, so you cannot deposit "for" somebody.

I see your point, but to me this doesn't make a big difference. A `depositFor` should still exist, as we do not know whether someone might need it (there are instances where multisigs just give an approval and a specific member can handle the deposit up to the allowance limit whenever they see fit from their own wallet), so I'd still add it just in case, personally.